blockchain/blockchain.go

```diff
@@ -575,3 +597,16 @@ func (bc *blockchain) emitToSubscribers(blk *block.Block) {
 	}
 	bc.pubSubManager.SendBlockToSubscribers(blk)
 }
+
+func (bc *blockchain) buildBlockTimeCalculator() (*blockutil.BlockTimeCalculator, error) {
+	return bc.btcBuilder.Clone().SetTipHeight(bc.TipHeight).SetHistoryBlockTime(func(height uint64) (time.Time, error) {
+		if height == 0 {
+			return time.Unix(bc.genesis.Timestamp, 0), nil
+		}
```

Comment:
after #4585, this `height == 0` check can move into dao?

Reply:
discussed offline